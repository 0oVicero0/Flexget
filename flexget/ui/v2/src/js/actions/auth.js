import * as utils from 'utils/actions';
import { get, post } from 'utils/fetch';

export const AUTH = 'AUTH';
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

const createAction = utils.createAction(AUTH);
const loading = utils.createLoading(AUTH);

export function checkLogin() {
  return (dispatch) => {
    get('/server/version')
      .then(() => dispatch(createAction(LOGIN)))
      .catch(() => {});
  };
}

export function login(credentials) {
  return (dispatch) => {
    dispatch(loading(LOGIN));
    post('/auth/login', credentials)
      .then(() => dispatch(createAction(LOGIN)))
      .catch(err => dispatch(createAction(LOGIN, err)));
  };
}

export function logout() {
  return (dispatch) => {
    dispatch(loading(LOGOUT));
    get('/auth/logout')
      .then(() => dispatch(createAction(LOGOUT)))
      .then(err => dispatch(createAction(LOGOUT, err)));
  };
}
