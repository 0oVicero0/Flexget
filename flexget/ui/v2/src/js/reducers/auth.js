import { LOGIN, LOGOUT } from 'actions/auth';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case LOGIN:
      return {
        loggedIn: true,
      };
    case LOGOUT:
      return {};
    default:
      if (action.error && action.payload && action.payload.status === 401) {
        return {};
      }

      return state;
  }
}
