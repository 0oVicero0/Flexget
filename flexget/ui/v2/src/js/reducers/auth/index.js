import { LOGIN, LOGOUT } from 'actions/auth';
import { ERROR_STATUS } from 'actions/status';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case LOGIN:
      return {
        loggedIn: true,
      };
    case LOGOUT:
      return {};
    case ERROR_STATUS:
      if (action.payload.statusCode === 401) {
        return {};
      }
    default: // eslint-disable-line no-fallthrough
      return state;
  }
}
