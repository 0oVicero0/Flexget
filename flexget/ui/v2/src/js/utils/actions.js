import { LOADING_STATUS } from 'actions/status';

export const createAction = (type, payload, {
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    message,
  },
});

export const loading = type => ({
  type: LOADING_STATUS,
  payload: {
    type,
  },
});
