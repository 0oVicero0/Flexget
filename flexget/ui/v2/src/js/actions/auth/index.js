import { createAction, loading } from 'utils/actions';
import { get, post } from 'utils/fetch';

const PREFIX = '@flexget/auth/';
export const LOGIN = `${PREFIX}LOGIN`;
export const LOGOUT = `${PREFIX}LOGOUT`;

export function checkLogin() {
  return dispatch => get('/server/version')
      .then(() => dispatch(createAction(LOGIN)))
      .catch(() => {});
}

export function login(credentials) {
  return (dispatch) => {
    dispatch(loading(LOGIN));
    return post('/auth/login', credentials)
      .then(() => dispatch(createAction(LOGIN)))
      .catch(err => dispatch(createAction(LOGIN, err)));
  };
}

export function logout() {
  return (dispatch) => {
    dispatch(loading(LOGOUT));
    return post('/auth/logout')
      .then(() => dispatch(createAction(LOGOUT)))
      .catch(err => dispatch(createAction(LOGOUT, err)));
  };
}
