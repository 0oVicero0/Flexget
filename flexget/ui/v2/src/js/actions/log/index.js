import oboe from 'oboe';
import { createAction } from 'utils/actions';

const PREFIX = '@flexget/log/';
export const LOG_START = `${PREFIX}LOG_START`;
export const LOG_MESSAGE = `${PREFIX}LOG_MESSAGE`;
export const LOG_DISCONNECT = `${PREFIX}LOG_DISCONNECT`;
export const LOG_LINES = `${PREFIX}LOG_LINES`;
export const LOG_QUERY = `${PREFIX}LOG_QUERY`;
export const LOG_CLEAR = `${PREFIX}LOG_CLEAR`;

export function startLogStream() {
  return (dispatch, getState) => {
    const { lines, query } = getState().log;
    const stream = oboe({
      url: `api/server/log?lines=${lines}&search=${query}`,
      method: 'GET',
    });

    const promise = new Promise((resolve, reject) => {
      stream
        .on('start', () => dispatch(createAction(LOG_START)))
        .on('node', '{message}', message => dispatch(createAction(LOG_MESSAGE, message)))
        .done(() => resolve())
        .fail(err => reject(err));
    });

    promise.abort = () => {
      stream.abort();
      dispatch(createAction(LOG_DISCONNECT));
    };

    return promise;
  };
}

export function setLines(lines) {
  return createAction(LOG_LINES, lines);
}

export function setQuery(query) {
  return createAction(LOG_QUERY, query);
}

export function clearLogs() {
  return createAction(LOG_CLEAR);
}

