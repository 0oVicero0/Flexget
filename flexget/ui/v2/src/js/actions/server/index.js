import { createAction, loading } from 'utils/actions';
import { post } from 'utils/fetch';

const PREFIX = '@flexget/server/';
export const SERVER_RELOAD = `${PREFIX}SERVER_RELOAD`;
export const SERVER_SHUTDOWN = `${PREFIX}SERVER_SHUTDOWN`;


function manageServer(action, operation) {
  return (dispatch) => {
    dispatch(loading(action));
    return post('/server/manage', { operation })
      .then(({ data }) => dispatch(createAction(action, {}, { message: data.message })))
      .catch(err => dispatch(createAction(action, err)));
  };
}

export function reloadServer() {
  return manageServer(SERVER_RELOAD, 'reload');
}

export function shutdownServer() {
  return manageServer(SERVER_SHUTDOWN, 'shutdown');
}
