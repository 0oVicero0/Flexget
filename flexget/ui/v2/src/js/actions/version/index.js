import { createAction, loading } from 'utils/actions';
import { get } from 'utils/fetch';

const PREFIX = '@flexget/version/';
export const GET_VERSION = `${PREFIX}GET_VERSION`;

export function getVersion() {
  return (dispatch) => {
    dispatch(loading(GET_VERSION));
    return get('/server/version')
      .then(({ data }) => dispatch(createAction(GET_VERSION, data)))
      .catch(err => dispatch(createAction(GET_VERSION, err)));
  };
}
