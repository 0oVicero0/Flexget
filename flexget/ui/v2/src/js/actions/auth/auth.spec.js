import fetchMock from 'fetch-mock';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import * as auth from 'actions/auth';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const credentials = {
  username: 'flexget',
  password: 'test',
};

describe('actions/auth', () => {
  afterEach(() => {
    fetchMock.restore();
  });

  describe('login', () => {
    it('dispatches the correct actions on succesful login', () => {
      const expectedActions = [
        expect.objectContaining({
          type: auth.LOGIN,
          meta: {
            namespace: auth.AUTH,
            loading: true,
          },
        }),
        expect.objectContaining({
          type: auth.LOGIN,
          error: false,
        }),
      ];
      const store = mockStore({ auth: {} });

      fetchMock
        .post('/api/auth/login', {});

      return store.dispatch(auth.login(credentials))
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
    });


    it('dispatches the correct actions on unsuccesful login', () => {
      const expectedActions = [
        expect.objectContaining({
          type: auth.LOGIN,
          meta: {
            namespace: auth.AUTH,
            loading: true,
          },
        }),
        expect.objectContaining({
          type: auth.LOGIN,
          error: true,
        }),
      ];
      const store = mockStore({ auth: {} });
      fetchMock
        .post('/api/auth/login', 401);

      return store.dispatch(auth.login(credentials))
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
    });
  });

  describe('checkLogin', () => {
    it('dispatches the login action if already logged in', () => {
      const expectedActions = [
        expect.objectContaining({
          type: auth.LOGIN,
          error: false,
        }),
      ];
      const store = mockStore({ auth: {} });

      fetchMock
        .get('/api/server/version', {});

      return store.dispatch(auth.checkLogin())
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
    });


    it('dispatches no actions if not logged in already', () => {
      const expectedActions = [];
      const store = mockStore({ auth: {} });
      fetchMock
        .get('/api/server/version', 401);

      return store.dispatch(auth.checkLogin())
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
    });
  });

  describe('logout', () => {
    it('dispatches the correct actions on succesful logout', () => {
      const expectedActions = [
        expect.objectContaining({
          type: auth.LOGOUT,
          meta: {
            namespace: auth.AUTH,
            loading: true,
          },
        }),
        expect.objectContaining({
          type: auth.LOGOUT,
          error: false,
        }),
      ];
      const store = mockStore({ auth: {} });

      fetchMock
        .get('/api/auth/logout', {});

      return store.dispatch(auth.logout())
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
    });


    it('dispatches the correct actions on unsuccesful logout', () => {
      const expectedActions = [
        expect.objectContaining({
          type: auth.LOGOUT,
          meta: {
            namespace: auth.AUTH,
            loading: true,
          },
        }),
        expect.objectContaining({
          type: auth.LOGOUT,
          error: true,
        }),
      ];
      const store = mockStore({ auth: {} });
      fetchMock
        .get('/api/auth/logout', 401);

      return store.dispatch(auth.logout())
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
    });
  });
});
