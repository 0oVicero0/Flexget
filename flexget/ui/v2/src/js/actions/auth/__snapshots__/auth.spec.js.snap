// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/auth checkLogin dispatches no actions if not logged in already 1`] = `Array []`;

exports[`actions/auth checkLogin dispatches the login action if already logged in 1`] = `
Array [
  Object {
    "error": false,
    "meta": Object {
      "message": undefined,
    },
    "payload": undefined,
    "type": "@flexget/auth/LOGIN",
  },
]
`;

exports[`actions/auth login dispatches the correct actions on successful login 1`] = `
Array [
  Object {
    "payload": Object {
      "type": "@flexget/auth/LOGIN",
    },
    "type": "LOADING_STATUS",
  },
  Object {
    "error": false,
    "meta": Object {
      "message": undefined,
    },
    "payload": undefined,
    "type": "@flexget/auth/LOGIN",
  },
]
`;

exports[`actions/auth login dispatches the correct actions on unsuccessful login 1`] = `
Array [
  Object {
    "payload": Object {
      "type": "@flexget/auth/LOGIN",
    },
    "type": "LOADING_STATUS",
  },
  Object {
    "error": true,
    "meta": Object {
      "message": undefined,
    },
    "payload": [Error],
    "type": "@flexget/auth/LOGIN",
  },
]
`;

exports[`actions/auth logout dispatches the correct actions on succesful logout 1`] = `
Array [
  Object {
    "payload": Object {
      "type": "@flexget/auth/LOGOUT",
    },
    "type": "LOADING_STATUS",
  },
  Object {
    "error": false,
    "meta": Object {
      "message": undefined,
    },
    "payload": undefined,
    "type": "@flexget/auth/LOGOUT",
  },
]
`;

exports[`actions/auth logout dispatches the correct actions on unsuccesful logout 1`] = `
Array [
  Object {
    "payload": Object {
      "type": "@flexget/auth/LOGOUT",
    },
    "type": "LOADING_STATUS",
  },
  Object {
    "error": true,
    "meta": Object {
      "message": undefined,
    },
    "payload": [Error],
    "type": "@flexget/auth/LOGOUT",
  },
]
`;
